src/handlers/mxcontrol.js

```diff
@@ -0,0 +1,159 @@
+                environmentErrors.map(async err => await msgMXControlRoom(err));
+
+            } else await msgMXControlRoom(`${targetName}.massexchange.com is **READY.**`);
+
+        } else await msgMXControlRoom("Please specify an environment");
+
+        return;
+    }
+
+    if (!useFullCLI &&
+        (upVerbs.includes(action) || rebootVerbs.includes(action)))
+        addEnvEntriesToDynamo(targetName);
+
+    MXControl.runTask(task).catch(err =>{
+        console.log(err);
+        msgMXControlRoom("Something went wrong with the operation. Ops has been notified.");
+    });
+};
+
+const addEnvEntriesToDynamo = async envName => {
+    return appNames.map(async appName =>
+        await mxDynamoDB.putItem({
+            "InstanceName": {"S":`${appName}-${envName}`}
+        }, dynamoName));
+};
+
+const msgMXControlRoom =
+    async message => hipchat.notify(message, {room: "MXControl"});
+
+const flattenStatus = statusJSON =>
+    Array.isArray(statusJSON[0])
+        ? statusJSON.reduce((agg, curr) => agg.concat(curr), [])
+        : statusJSON;
+
+const formatStatusResponse = statusJSON =>
+    flattenStatus(statusJSON).map(parseStatusText);
+
+const parseStatusText = ({
+    InstanceName,
+    InstanceState,
+    InstanceSize,
+    InstanceAddress
+}) =>`## ${InstanceName}
+${InstanceState}
+${InstanceSize}
+${InstanceAddress}
+`;
+
+const checkForControlTaskErrors = async MXControlTask => {
+    const tasksWithErrors = await MXControl.getControlTaskErrors(MXControlTask);
+    if (tasksWithErrors.length != 0) { //For now, just assume one task will ever exist
+        const errorArray = parseMXControlErrors(tasksWithErrors[0].errors);
+        await msgMXControlRoom("## Errors:");
+        await msgMXControlRoom(errorArray.join("\n"));
+        await msgMXControlRoom("## Please double check your input, and try again.");
+        return false;
+    }
+    return true;
+};
+
+
+const parseMXControlErrors = errorArray =>
+    errorArray.reduce((agg, curr) => agg.concat(curr), []);
+
+const checkEnvironmentReadiness = async(statusJSON, envName) => {
+    const errors = [];
+
+    const isEnvironmentMember = obj =>
+        obj.InstanceEnvironment == envName
+        || obj.InstanceName == `mxenvironment-${envName}`;
+
+    const flatStatus = flattenStatus(statusJSON).filter(isEnvironmentMember);
+
+    if (flatStatus.length == 0) {
+        errors.push(`Environment ${envName} doesn't exist!`);
+        return errors;
+    }
+
+    const backend = flatStatus.filter(obj =>
+        obj.InstanceApplication == "MXBackend")[0];
+
+    const frontend = flatStatus.filter(obj =>
+        obj.InstanceApplication == "MXWeb")[0];
+
+    const db = flatStatus.filter(obj =>
+        obj.InstanceName == `mxenvironment-${envName}`)[0];
+
+    const addressFromDNS = await dnsLookup(`${envName}.massexchange.com`);
+
+    if (backend.InstanceState != "running" && frontend.InstanceState != "running")
```

Comment:
you can extract these checks into variables, as you repeat them many times

Reply:
Didn't feel the need to alias them, but okay.